web/client/components/TOC/fragments/settings/VisibilityLimitsForm.jsx

```diff
@@ -0,0 +1,405 @@
+            }}
+            onChange={(option) => {
+                const newValue = option?.value && parseFloat(option.value);
+                onChange(isNaN(newValue) ? undefined : { ...option, value: newValue });
+            }}
+        />
+    );
+}
+/**
+ * Form to set min and max resolutions values from layer object
+ * @memberof components.TOC
+ * @name VisibilityLimitsForm
+ * @class
+ * @prop {node|string} title
+ * @prop {object} layer
+ * @prop {number} zoom
+ * @prop {string} projection projection code of the target map
+ * @prop {array} resolutions resolutions array of the target map
+ * @prop {string} defaultLimitsType default type for visibility limits 'scale' or 'resolution'
+ * @prop {number} dpi dpi value of the browser
+ * @prop {function} onChange callback triggered after changing the form
+ */
+function VisibilityLimitsForm({
+    title,
+    layer,
+    zoom,
+    projection,
+    resolutions = getResolutions(),
+    defaultLimitsType,
+    limitsTypesOptions,
+    dpi,
+    onChange
+}) {
+
+    const [limitsType, setLimitsType] = useState(defaultLimitsType || limitsTypesOptions[0].value);
+
+    const {
+        maxResolution,
+        minResolution,
+        disableResolutionLimits
+    } = layer;
+
+    const dpu = dpi2dpu(dpi, projection);
+
+    function updateOptions(options, newValue) {
+        const values = options.map((option) => option[limitsType]);
+        const isMissing = !isNil(newValue[limitsType]) && values.indexOf(newValue[limitsType]) === -1;
+        return isMissing
+            ? [ newValue, ...options].sort((a, b) => a[limitsType] > b[limitsType] ? -1 : 1)
+            : options;
+    }
+
+    function getResolutionObject(value, type) {
+        if (type === 'scale') {
+            const resolution = value / dpu;
+            return {
+                resolution: resolution,
+                scale: value,
+                zoom: getZoomFromResolution(resolution, resolutions)
+            };
+        }
+        return {
+            resolution: value,
+            scale: value * dpu,
+            zoom: getZoomFromResolution(value, resolutions)
+        };
+    }
+
+    const [options, setOptions] = useState([]);
+    const [maxValue, setMaxValue] = useState(maxResolution && {
+        resolution: maxResolution,
+        scale: maxResolution * dpu,
+        zoom: getZoomFromResolution(maxResolution, resolutions)
+    });
+    const [minValue, setMinValue] = useState(minResolution && {
+        resolution: minResolution,
+        scale: minResolution * dpu,
+        zoom: getZoomFromResolution(minResolution, resolutions)
+    });
+    const [loading, setLoading] = useState(false);
+    const [message, setMessage] = useState();
+    const [error, setError] = useState();
+    const isMounted = useRef();
+
+    useEffect(() => {
+        isMounted.current = true;
+        return () => {
+            isMounted.current = false;
+        };
+    }, []);
+
+    function clearAlters() {
```

Comment:
my 2 cents on this name
```suggestion
    function clearMessages() {
```
or
```suggestion
    function clearAlerts() {
```